lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -247,7 +247,10 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
 
     * `:row_fetcher` - Required. A function which receives the params and the node and
       returns a tuple with the rows and the total number:
-      `(params(), node()) -> {list(), integer() | binary()}`
+      ```
+      (params(), node()) -> {list(), integer() | binary()} |
+        (params(), node(), socket()) -> {list(), integer() | binary(), socket()}
+      ```
```

Comment:
Suggestion: I would change it to `row_fetcher/3` and no longer support `row_fetcher/2`. I would also call the last argument `state`, as it doesn't have to be a socket? :)

Reply:
Are you sure you want to drop `row_fetcher/2` support? It will break user's custom pages. I know we aren't in a 1.0.0. yet, but we could warning the user for a couple of releases. WDYT?

I like the change to use a generic `state` instead of the `socket`. Should we add an optional `:row_fetcher_initial_state` field in `table/1` function to be passed the first time? Or should it be handled by `row_fetcher/3` when receiving `nil`?

Reply:
Ah, good questions. So what if we support:

    row_fetcher/2 and {row_fetcher/3, initial_state}?

Reply:
Do you care if we support `row_fetcher/2 | row_fetcher/3 | {row_fetcher/3, initial_state}`? Not a strong opinion, but I prefer it :) Should I warning the `row_fetcher/2` usage?

Reply:
I think you pointed out that the row_fetcher/3 needs the initial state and I would rather not hardcode it to the socket, so I would prefer to not support row_fetcher/3 without the tuple. :)